### PR TITLE
Pull Kyber division fixes from PQ-Crystals into main

### LINKED ISSUE #1645: Further Kyber division fixes
A few more potentially non-constant-time divisions have been identified in Kyber:
https://github.com/pq-crystals/kyber/commit/272125f6acc8e8b6850fd68ceb901a660ff48196

### PR DESCRIPTION
**Pull request by praveksharma**
Pull recent fixes made to Kyber from [PQ-Crystals](https://github.com/pq-crystals/kyber/commit/272125f6acc8e8b6850fd68ceb901a660ff48196) into main branch and patches Kyber aarch64 implementation.

Fixes #1645 (see also #1652).

* [No] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [No] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in fully supported downstream projects dependent on these, i.e., [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) and [OQS-OpenSSH](https://github.com/open-quantum-safe/openssh) will also need to be ready for review and merge by the time this is merged.)

### CODE REVIEW

File: scripts/copy_from_upstream/copy_from_upstream.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ upstreams:
     name: pqcrystals-kyber
     git_url: https://github.com/pq-crystals/kyber.git
     git_branch: master
-    git_commit: dda29cc63af721981ee2c831cf00822e69be3220
```
Comment (bhess) [inline scripts/copy_from_upstream/copy_from_upstream.yml:28]: The latest commit upstream is `b628ba78711bc28327dc7d2d5c074a00f061884e`. Perhaps pull this version?

**Review by bhess — COMMENTED** [review]
Thanks for these updates @praveksharma & @SWilson4.

**Comment by dstebila** [comment]
I checked with @cryptojedi and he says that after this there shouldn't be any operations of the form `/KYBER_Q` left in the source code -- we could consider a CI test that grep's for those. He also writes that one could compile with `-Os` and then check if the output has an DIV instructions.

**Review by dstebila — APPROVED** [review]

File: docs/algorithms/kem/kyber.md [hunk 2]
```diff
@@ -7,9 +7,9 @@
 - **Authors' website**: https://pq-crystals.org/
 - **Specification version**: NIST Round 3 submission.
 - **Primary Source**<a name="primary-source"></a>:
-  - **Source**: https://github.com/pq-crystals/kyber/commit/dda29cc63af721981ee2c831cf00822e69be3220 with copy_from_upstream patches
+  - **Source**: https://github.com/pq-crystals/kyber/commit/b628ba78711bc28327dc7d2d5c074a00f061884e with copy_from_upstream patches
   - **Implementation license (SPDX-Identifier)**: CC0-1.0 or Apache-2.0
-- **Optimized Implementation sources**: https://github.com/pq-crystals/kyber/commit/dda29cc63af721981ee2c831cf00822e69be3220 with copy_from_upstream patches
+- **Optimized Implementation sources**: https://github.com/pq-crystals/kyber/commit/b628ba78711bc28327dc7d2d5c074a00f061884e with copy_from_upstream patches
   - **oldpqclean-aarch64**:<a name="oldpqclean-aarch64"></a>
       - **Source**: https://github.com/PQClean/PQClean/commit/8e220a87308154d48fdfac40abbb191ac7fce06a with copy_from_upstream patches
```
Comment (baentsch) [inline docs/algorithms/kem/kyber.md:14]: Is there a better solution for this? There's apparently no trace (at least in the documentation) as to whether the (aarch64) code version that `liboqs` runs does or does not contain the "DIV" fix. Code review shows it does, but a more simple inspection by checking these alg documentation files does not. It's "hidden" within the statement (and a changed file contents of) "copy_from_upstream patches". Maybe worth while adding a hash of those patches here at some time?

**Review by baentsch — APPROVED** [review]
Generally LGTM - just some "musing" on documentation as per comment.

**Review by bhess — APPROVED** [review]

**Pull request merged**